--- libavcodec/ass.h.bak	2020-05-21 22:33:14.000000000 +0900
+++ libavcodec/ass.h	2020-05-19 23:54:13.282164200 +0900
@@ -32,15 +32,18 @@
  * @name Default values for ASS style
  * @{
  */
-#define ASS_DEFAULT_FONT        "Arial"
-#define ASS_DEFAULT_FONT_SIZE   16
-#define ASS_DEFAULT_COLOR       0xffffff
-#define ASS_DEFAULT_BACK_COLOR  0
-#define ASS_DEFAULT_BOLD        0
-#define ASS_DEFAULT_ITALIC      0
-#define ASS_DEFAULT_UNDERLINE   0
-#define ASS_DEFAULT_ALIGNMENT   2
-#define ASS_DEFAULT_BORDERSTYLE 1
+#define ASS_DEFAULT_FONT        	"Microsoft YaHei"
+#define ASS_DEFAULT_FONT_SIZE   	24
+#define ASS_DEFAULT_COLOR       	0xffffff
+#define ASS_DEFAULT_PRIMARY_COLOR	0xffffff
+#define ASS_DEFAULT_SECONDARY_COLOR	0xf0000000
+#define ASS_DEFAULT_OUTLINE_COLOR	0x492704
+#define ASS_DEFAULT_BACK_COLOR		0x31492704
+#define ASS_DEFAULT_BOLD        	0
+#define ASS_DEFAULT_ITALIC      	0
+#define ASS_DEFAULT_UNDERLINE   	0
+#define ASS_DEFAULT_ALIGNMENT   	2
+#define ASS_DEFAULT_BORDERSTYLE 	1
 /** @} */
 
 typedef struct FFASSDecoderContext {
@@ -49,6 +52,34 @@
 
 /**
  * Generate a suitable AVCodecContext.subtitle_header for SUBTITLE_ASS.
+ * Can specify all fields explicitly
+ *
+ * @param avctx pointer to the AVCodecContext
+ * @param play_res_x subtitle frame width
+ * @param play_res_y subtitle frame height
+ * @param font name of the default font face to use
+ * @param font_size default font size to use
+ * @param primary_color default text color to use (ABGR)
+ * @param secondary_color default secondary text color to use (ABGR)
+ * @param outline_color default outline color to use (ABGR)
+ * @param back_color default background color to use (ABGR)
+ * @param bold 1 for bold text, 0 for normal text
+ * @param italic 1 for italic text, 0 for normal text
+ * @param underline 1 for underline text, 0 for normal text
+ * @param border_style 1 for outline, 3 for opaque box
+ * @param alignment position of the text (left, center, top...), defined after
+ *                  the layout of the numpad (1-3 sub, 4-6 mid, 7-9 top)
+ * @return >= 0 on success otherwise an error code <0
+ */
+int ff_ass_subtitle_header_full(AVCodecContext *avctx,
+                                int play_res_x, int play_res_y,
+                                const char *font, int font_size,
+                                int primary_color, int secondary_color,
+                                int outline_color, int back_color,
+                                int bold, int italic, int underline,
+                                int border_style, int alignment);
+/**
+ * Generate a suitable AVCodecContext.subtitle_header for SUBTITLE_ASS.
  *
  * @param avctx pointer to the AVCodecContext
  * @param font name of the default font face to use
